openedx/adg/lms/webinars/views.py

```diff
@@ -0,0 +1,58 @@
+"""
+All views for webinars app
+"""
+from django.contrib.auth.decorators import login_required
+from django.http import HttpResponseServerError
+from django.shortcuts import get_object_or_404, redirect, render
+from django.urls import reverse
+from django.utils.decorators import method_decorator
+from django.utils.translation import ugettext as _
+from django.views import View
+from django.views.decorators.csrf import ensure_csrf_cookie
+
+from .helpers import send_webinar_registration_email
+from .models import Webinar, WebinarRegistration
+
+
+def webinar_description_page_view(request, pk):
+    # TODO This is a temp view and will be properly functional in LP-2623
+    return render(request, 'adg/lms/webinar/description_page.html', context={'pk': pk})
+
+
+class WebinarRegistrationView(View):
+    """
+    A view to register user in a webinar event or cancel already registered user in a webinar.
+    """
+
+    @method_decorator(login_required)
+    @method_decorator(ensure_csrf_cookie)
+    def post(self, request, pk, action):
+        """
+        Register in a webinar or cancel registration
+        Args:
+            request (HttpRequest): HTTP request
+            pk (int): Webinar id, the primary key of model
+            action (str): Actions to perform on event i.e. `register` or `cancel`
+        Returns:
+            HttpResponse: On success redirect to webinar page otherwise through HTTP error code as per nature
+            of error i.e. 404 or 500
+        """
+        webinar = get_object_or_404(Webinar, pk=pk)
```

Comment:
Kindly take a look at [SingleObjectMixin](https://docs.djangoproject.com/en/3.1/topics/class-based-views/mixins/#using-singleobjectmixin-with-view), I think that will clean up some of our code.

Reply:
done